articles/react-native-firebase-text-recognition/index.md

```diff
@@ -0,0 +1,429 @@
+
+After the package is installed, import the `launchCamera` and `launchImageLibrary` functions from the package.
+
+```JSX
+import { launchCamera, launchImageLibrary } from 'react-native-image-picker';
+```
+
+Both functions accept 2 arguments. The first argument is `options` for the camera or the gallery, and the second argument is a callback function. This callback function is called when the user picks an image or cancels the operation.
+
+Check out the [API Reference](https://www.npmjs.com/package/react-native-image-picker#api-reference) for more details about these functions.
+
+Now let's add 2 functions, one for each button.
+
+```JSX
+const onTakePhoto = () => launchCamera({ mediaType: 'image' }, onImageSelect);
+
+const onSelectImagePress = () => launchImageLibrary({ mediaType: 'image' }, onImageSelect);
+```
+
+Let's create a function called `onImageSelect`. This is the callback function that we are passing to the `launchCamera` and the `launchImageLibrary` functions. We will get the details of the image that the user picked in this callback function.
+
+We should start the text recognition only when the user did not cancel the media picker. If the user canceled the operation, the picker will send a `didCancel` property in the response object.
+
+```JSX
+const onImageSelect = async (media) => {
+  if (!media.didCancel) {
+    // Text Recognition Process
+  }
+};
+```
+
+You can learn more about the response object that we get from the `launchCamera` and the `launchImageLibrary` functions [here](https://www.npmjs.com/package/react-native-image-picker#the-response-object).
+
+Now, pass these functions to the `onPress` prop of the `TouchableOpacity` for the respective buttons.
+
+```JSX
+<View>
+  <TouchableOpacity style={styles.button} onPress={onTakePhoto}>
+    <Text style={styles.buttonText}>Take Photo</Text>
+  </TouchableOpacity>
+  <TouchableOpacity style={styles.button} onPress={onSelectImagePress}>
+    <Text style={styles.buttonText}>Pick a Photo</Text>
+  </TouchableOpacity>
+<View>
+```
+
+Let's create a state to display the selected image on the UI.
+
+```JSX
+const [image, setImage] = useState();
+```
+
+Now, let's add an Image component below the buttons to display the selected image.
+
+```JSX
+<View>
+  <TouchableOpacity style={styles.button} onPress={onTakePhoto}>
+    <Text style={styles.buttonText}>Take Photo</Text>
+  </TouchableOpacity>
+  <TouchableOpacity style={styles.button} onPress={onSelectImagePress}>
+    <Text style={styles.buttonText}>Pick a Photo</Text>
+  </TouchableOpacity>
+  <Image source={{uri: image}} style={styles.image} resizeMode="contain" />
+</View>
+```
+
+Styles for the Image:
+
+```JSX
+image: {
+  height: 300,
+  width: 300,
+  marginTop: 30,
+  borderRadius: 10,
+},
+```
+
+Let's set the image state with the URI of the selected image in the `onImageSelect` function when the user did not cancel the operation.
```

Comment:
I think it would be better to say:

If the user did not cancel the operation, we go ahead and set the image state with the URI of the selected image in the `onImageSelect` function.

Reply:
agree. fixed.